app/smc/sysbuild.cmake

```diff
@@ -38,123 +52,92 @@ ExternalZephyrProject_Add(
 	BUILD_ONLY 1
 )
 
-ExternalZephyrProject_Add(
-	APPLICATION dmc
-	SOURCE_DIR  ${APP_DIR}/../dmc
-	BOARD       ${SB_CONFIG_DMC_BOARD}
-	BUILD_ONLY 1
-)
-
-if(BOARD STREQUAL "tt_blackhole")
-  # Map board revision names to folder names for spirom config data
-  string(TOUPPER ${BOARD_REVISION} BASE_NAME)
-  set(PROD_NAME "${BASE_NAME}-1")
-elseif(BOARD STREQUAL "native_sim")
-  # Use P100 data files to stand in
-  set(PROD_NAME "P100-1")
-  set(BOARD_REVISION "p100")
-else()
-  message(FATAL_ERROR "No support for board ${BOARD}")
+# galaxy does not have dmc image in SPI
+if(NOT "${BOARD_REVISION}" STREQUAL "galaxy")
+  ExternalZephyrProject_Add(
+    APPLICATION dmc
+    SOURCE_DIR  ${APP_DIR}/../dmc
+    BOARD       ${SB_CONFIG_DMC_BOARD}
+    BUILD_ONLY 1
+  )
 endif()
 
-set(OUTPUT_BOOTFS ${CMAKE_BINARY_DIR}/tt_boot_fs.bin)
+# ======== Defines for filesystem generation ========
 set(OUTPUT_FWBUNDLE ${CMAKE_BINARY_DIR}/update.fwbundle)
 
 set(DMC_OUTPUT_BIN ${CMAKE_BINARY_DIR}/dmc/zephyr/zephyr.bin)
 set(SMC_OUTPUT_BIN ${CMAKE_BINARY_DIR}/${DEFAULT_IMAGE}/zephyr/zephyr.bin)
 set(RECOVERY_OUTPUT_BIN ${CMAKE_BINARY_DIR}/recovery/zephyr/zephyr.bin)
 
-# Generate filesystem
+if (PROD_NAME MATCHES "^GALAXY")
+  set(BOOTFS_DEPS ${SMC_OUTPUT_BIN} ${RECOVERY_OUTPUT_BIN})
+else()
+  set(BOOTFS_DEPS ${DMC_OUTPUT_BIN} ${SMC_OUTPUT_BIN} ${RECOVERY_OUTPUT_BIN})
+endif()
 
+# # ======== Helper functions for generate filesystem ========
+function(add_bootfs_and_fwbundle bootfs_yaml output_bootfs output_fwbundle prod_name)
+  # Remaining arguments are bootfs_deps
+  set(bootfs_deps ${ARGN})
+
+  add_custom_command(OUTPUT ${output_bootfs}
+    COMMAND ${PYTHON_EXECUTABLE}
+    ${APP_DIR}/../../scripts/tt_boot_fs.py mkfs
+    ${bootfs_yaml}
+    ${output_bootfs}
+    --build-dir ${CMAKE_BINARY_DIR}
+    DEPENDS ${bootfs_deps})
+
+  # Generate firmware bundle that can be used to flash this build on a board
+  # using tt-flash
+  add_custom_command(OUTPUT ${output_fwbundle}
+    COMMAND ${PYTHON_EXECUTABLE}
+    ${APP_DIR}/../../scripts/tt_boot_fs.py fwbundle
+    -v "${BUNDLE_VERSION_STRING}"
+    -o ${output_fwbundle}
+    ${prod_name}
+    ${output_bootfs}
+    DEPENDS ${output_bootfs})
+endfunction()
```

Comment:
This is good. We should put the cmake function into cmake/something.cmake and include that file here.